ggml/src/ggml-backend.c

```diff
@@ -1124,14 +1120,8 @@ static int ggml_backend_sched_backend_from_buffer(ggml_backend_sched_t sched, co
     return -1;
 }
 
-#if 0
-static char causes[GGML_DEFAULT_GRAPH_SIZE*16 + GGML_SCHED_MAX_SPLITS*GGML_SCHED_MAX_SPLIT_INPUTS][128]; // debug only
-#define SET_CAUSE(node, ...) sprintf(causes[hash_id(node)], __VA_ARGS__)
-#define GET_CAUSE(node) causes[hash_id(node)]
-#else
```

Comment:
This should not be removed completely, it is useful for debugging. Just use some large value for `GGML_SCHED_MAX_SPLITS`.

Reply:
Big thanks for the review. I fixed and readded debug code for causes in [fe40950](https://github.com/ggerganov/llama.cpp/pull/9047/commits/fe40950b85b1ea203aad695e741133ae367101fd). For the sake of readability, I introduced GGML_SCHED_MAX_SPLITS_DEBUG. I choose 4096 as this should be enough for every model, I'm aware of.